src/wara_ps_client.cc

```diff
@@ -10,39 +10,46 @@
 #include <boost/uuid/uuid_io.hpp>
 #include <utility>
 
-#define QOS_AT_LEAST_ONCE 1
+using namespace std::chrono;
+using namespace std::string_view_literals;
+
+constexpr bool QOS_AT_LEAST_ONCE{true};
 
 using json = nlohmann::json;
 
-const bool kRetain = false;
-const std::string TOPIC_PREFIX = "waraps/unit/ground/real/ljudkriget/";
+constexpr bool kRetain = false;
+constexpr auto TOPIC_PREFIX = "waraps/unit/ground/real/ljudkriget/"sv;
 
 std::string WaraPSClient::GenerateUUID() {
     boost::uuids::uuid uuid = boost::uuids::random_generator()();
     return boost::uuids::to_string(uuid);
 }
 
-std::string WaraPSClient::GenerateFullTopic(const std::string &topic) {
-    return TOPIC_PREFIX + topic;
+std::string WaraPSClient::GenerateFullTopic(std::string_view topic) {
+    std::string str;
+    str.reserve(TOPIC_PREFIX.length() + topic.length());
+    str += TOPIC_PREFIX;
+    str += topic;
+    return str;
 }
 
 std::string WaraPSClient::GenerateHeartBeatMessage() const {
     json j = {
-            {"agent-type", "surface"},
-            {"agent-uuid", kUUID},
-            {"levels",     {"sensor", "direct execution"}},
-            {"name",       "ljudkriget"},
-            {"rate",       heartbeat_interval.count() / 1000},
-            {"stamp",      (double) std::chrono::system_clock::now().time_since_epoch().count() / 1000.0},
-            {"type",       "HeartBeat"}};
+        {"agent-type", "surface"},
+        {"agent-uuid", kUUID},
+        {"levels",     {"sensor", "direct execution"}},
+        {"name",       "ljudkriget"},
+        {"rate",       duration_cast<milliseconds>(heartbeat_interval).count()},
```

Comment:
You likely want microseconds here.

Reply:
Isn't UNIX Epoch time usually in milliseconds? I might be misremembering.